connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorHeartbeatConnector.java

```diff
@@ -50,6 +50,10 @@ public void stop() {
 
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
+        // if emitting heartbeat is disabled
+        if (config.emitHeartbeatsInterval().isNegative()) {
```

Comment:
Should we check for `!config.enabled()` here too?

Reply:
No, we want to emit heartbeats even if A->B.enabled=false (see my earlier comment). Basically, we want heartbeats to be sent even if replication is disabled, _unless_ heartbeats are explicitly disabled with emit.heartbeats.enabled = false.

Reply:
Gotcha, maybe worth adding that the explanation to the comment

Reply:
Great, I will add the the explanation to the comment in the next udpate

Reply:
if `isNegative()` is not highly recommended, what about this change? Should we still check `isNegative()` or return the interval as `Duration.ofMillis(Long.MAX_VALUE)` ?